src/main/java/org/fcrepo/audit/InternalAuditor.java

```diff
@@ -177,42 +208,76 @@ public void releaseConnections() {
-        auditNode.setProperty(PREMIS_OBJ, uri, PropertyType.URI);
-        auditNode.setProperty(PREMIS_AGENT, new String[]{userID,userAgent});
-        if (auditEventType != null) {
-            auditNode.setProperty(PREMIS_TYPE, auditEventType, PropertyType.URI);
+    public void createAuditNode(final FedoraEvent event) throws RepositoryRuntimeException, IOException {
+        try {
+            final String uuid = pidMinter.mintPid();
+            final String userData = event.getUserData();
+            final ObjectMapper mapper = new ObjectMapper();
+            final JsonNode json = mapper.readTree(userData);
+            final String userAgent = json.get("userAgent").asText();
+            String baseURL = json.get("baseURL").asText();
+            if (baseURL.endsWith("/")) {
+                baseURL = baseURL.substring(0, baseURL.length() - 1);
+            }
+            String path = event.getPath();
+            if ( path.endsWith("/" + JCR_CONTENT) ) {
+                path = path.replaceAll("/" + JCR_CONTENT,"");
+            }
+            final String uri = baseURL + path;
+            final Long timestamp =  event.getDate();
+            final DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+            df.setTimeZone(TimeZone.getTimeZone("UTC"));
+            final String eventDate = df.format(new Date(timestamp));
+            final String userID = event.getUserID();
+            final String eventType = getEventURIs(event.getTypes());
+            final String properties = Joiner.on(',').join(event.getProperties());
+            final String auditEventType = getAuditEventType(eventType, properties);
+            final FedoraResource auditResource = containerService.findOrCreate(session,
+                    AUDIT_CONTAINER_LOCATION + "/" + uuid);
+
+            LOGGER.debug("Audit node {} created for event.", uuid);
+
+            //auditResource.addMixin("fedora:Resource");
+            final Model m = createDefaultModel();
+            final Resource s = createResource(baseURL + AUDIT_CONTAINER_LOCATION + "/" + uuid);
+            m.add(createStatement(s, RDF_TYPE, createResource(INTERNAL_EVENT)));
+            m.add(createStatement(s, RDF_TYPE, createResource(PREMIS_EVENT)));
+            m.add(createStatement(s, RDF_TYPE, createResource(PROV_EVENT)));
+            m.add(createStatement(s, PREMIS_TIME, createTypedLiteral(eventDate, XSDdateTime)));
+            m.add(createStatement(s, PREMIS_AGENT, createTypedLiteral(userID, XSDstring)));
+            m.add(createStatement(s, PREMIS_AGENT, createTypedLiteral(userAgent, XSDstring)));
+            if (auditEventType != null) {
+                m.add(createStatement(s, PREMIS_TYPE, createResource(auditEventType)));
+            }
+
+            final HttpResourceConverter subjects = new HttpResourceConverter(session, UriBuilder.fromUri(baseURL));
+            auditResource.replaceProperties(subjects, m, new RdfStream(), containerService);
```

Comment:
This isn't working for some reason -- I'm digging around in the RDF machinery trying to figure out why, but no luck so far.  I did fix the HttpResourceConvertor to use the audit resource URI for UriBuilder.forUri(), but that didn't fix it.